docs/decisions/0071-multi-agent-orchestration.md

```diff
@@ -0,0 +1,620 @@
+        AG0[agent 0]
+      end
+      subgraph AA1[Agent Actor]
+        AG1[agent 1]
+      end
+
+    end
+
+    IT[Internal Topic]
+
+    OA[Optional Actor]
+
+  end
+
+  %% Connections
+
+  AA0 <-.Direct messaging.-> AA1
+  AA0 <-.Direct messaging.-> OA
+  AA1 <-.Direct messaging.-> OA
+
+  IT <-.Broadcast.-> AA0
+  IT <-.Broadcast.-> AA1
+  IT <-.Broadcast.-> OA
+```
+
+#### Agent Actor
+
+This is a wrapper around a Semantic Kernel agent so that the agent can send and receive messages from the runtime. The `AgentActorBase` will inherit the [`RoutedAgent`](https://microsoft.github.io/autogen/stable/reference/python/autogen_core.html#autogen_core.RoutedAgent) class:
+
+```python
+class AgentActorBase(RoutedAgent):
+    """A agent actor for multi-agent orchestration running on Agent runtime."""
+
+    def __init__(self, agent: Agent) -> None:
+        """Initialize the agent container.
+
+        Args:
+            agent (Agent): An agent to be run in the container.
+        """
+        self._agent = agent
+        self._agent_thread = None
+        # Chat history to temporarily store messages before the agent thread is created
+        self._chat_history = ChatHistory()
+
+        RoutedAgent.__init__(self, description=agent.description or "Semantic Kernel Agent")
+```
+
+Orchestrations will have their own agent actor that extends the `AgentActorBase` because each orchestration can have its own set of message handlers.
+
+> To learn more about messages and message handlers, please refer to the [AutoGen documentation](https://microsoft.github.io/autogen/stable/user-guide/core-user-guide/framework/message-and-communication.html).
+
+For example, for the group chat orchestration, the agent actor will look like this:
+
+```python
+class GroupChatAgentActor(AgentActorBase):
+    """An agent actor for agents that process messages in a group chat."""
+
+    @message_handler
+    async def _handle_start_message(self, message: GroupChatStartMessage, ctx: MessageContext) -> None:
+        """Handle the initial message(s) provided by the user."""
+        ...
+
+    @message_handler
+    async def _handle_response_message(self, message: GroupChatResponseMessage, ctx: MessageContext) -> None:
+        """Handle the response message from other agents in the group chat."""
+        ...
+
+    @message_handler
+    async def _handle_request_message(self, message: GroupChatRequestMessage, ctx: MessageContext) -> None:
+        """Handle the request message from the group manager."""
+        ...
+```
+
+Agent actors in other orchestrations will handle different message types or different number of message types. This proposal doesn't make any restrictions on how agent actors interact with each other inside an orchestration, i.e. rules are defined by individual orchestrations.
+
+#### Data Transform Logic
+
+The signature of the data transform logic will be as follows:
+
+```python
+DefaultTypeAlias = Union[ChatMessageContent, list[ChatMessageContent]]
```

Comment:
I know this isn't a specific code review, but can't the be `ChatMessageContent | list[ChatMessageContent]` since we support Py >=3.10? :)